drivers/sx1276/sx1276.c

```diff
@@ -0,0 +1,1302 @@
+
+            sx1276_reg_write(dev, SX1276_REG_DIOMAPPING1, 0x00);
+            //sx1276_reg_write(dev, SX1276_REG_DIOMAPPING2, 0x20); /* DIO5=mode_ready */
+            break;
+        default:
+            break;
+    }
+}
+
+#define RXLORA_RXMODE_RSSI_SX1276_REG_MODEM_CONFIG1 0x0A
+#define RXLORA_RXMODE_RSSI_SX1276_REG_MODEM_CONFIG2 0x70
+
+uint32_t sx1276_random(sx1276_t *dev)
+{
+    uint8_t i;
+    uint32_t rnd = 0;
+
+    sx1276_set_modem(dev, SX1276_MODEM_LORA); /* Set LoRa modem ON */
+
+    /* Disable LoRa modem interrupts */
+    sx1276_reg_write(dev, SX1276_REG_LR_IRQFLAGSMASK, SX1276_RF_LORA_IRQFLAGS_RXTIMEOUT |
+                     SX1276_RF_LORA_IRQFLAGS_RXDONE |
+                     SX1276_RF_LORA_IRQFLAGS_PAYLOADCRCERROR |
+                     SX1276_RF_LORA_IRQFLAGS_VALIDHEADER |
+                     SX1276_RF_LORA_IRQFLAGS_TXDONE |
+                     SX1276_RF_LORA_IRQFLAGS_CADDONE |
+                     SX1276_RF_LORA_IRQFLAGS_FHSSCHANGEDCHANNEL |
+                     SX1276_RF_LORA_IRQFLAGS_CADDETECTED);
+
+    sx1276_set_op_mode(dev, SX1276_RF_OPMODE_STANDBY);
+    sx1276_reg_write(dev, SX1276_REG_LR_MODEMCONFIG1, RXLORA_RXMODE_RSSI_SX1276_REG_MODEM_CONFIG1);
+    sx1276_reg_write(dev, SX1276_REG_LR_MODEMCONFIG1, RXLORA_RXMODE_RSSI_SX1276_REG_MODEM_CONFIG2);
+
+    /* Set radio in continuous reception */
+    sx1276_set_op_mode(dev, SX1276_RF_OPMODE_RECEIVER);
+
+    for (i = 0; i < 32; i++) {
+        xtimer_usleep(1000); /* wait for the chaos */
+
+        /* Non-filtered RSSI value reading. Only takes the LSB value */
+        rnd |= ((uint32_t) sx1276_reg_read(dev, SX1276_REG_LR_RSSIWIDEBAND) & 0x01) << i;
+    }
+
+    sx1276_set_sleep(dev);
+
+    return rnd;
+}
+
+/**
+ * @brief Performs the Rx chain calibration for LF and HF bands
+ * @note Must be called just after the reset so all registers are at their
+ *         default values
+ */
+static void _rx_chain_calibration(sx1276_t *dev)
+{
+    uint8_t reg_pa_config_init_val;
+    uint32_t initial_freq;
+
+    /* Save context */
+    reg_pa_config_init_val = sx1276_reg_read(dev, SX1276_REG_PACONFIG);
+    initial_freq = (double) (((uint32_t) sx1276_reg_read(dev, SX1276_REG_FRFMSB) << 16)
+                             | ((uint32_t) sx1276_reg_read(dev, SX1276_REG_FRFMID) << 8)
+                             | ((uint32_t) sx1276_reg_read(dev, SX1276_REG_FRFLSB))) * (double) SX1276_FREQ_STEP;
+
+    /* Cut the PA just in case, RFO output, power = -1 dBm */
+    sx1276_reg_write(dev, SX1276_REG_PACONFIG, 0x00);
+
+    /* Launch Rx chain calibration for LF band */
+    sx1276_reg_write(dev,
+                     SX1276_REG_IMAGECAL,
+                     (sx1276_reg_read(dev, SX1276_REG_IMAGECAL) & SX1276_RF_IMAGECAL_IMAGECAL_MASK)
+                     | SX1276_RF_IMAGECAL_IMAGECAL_START);
+
+    while ((sx1276_reg_read(dev, SX1276_REG_IMAGECAL) & SX1276_RF_IMAGECAL_IMAGECAL_RUNNING)
+           == SX1276_RF_IMAGECAL_IMAGECAL_RUNNING) {
```

Comment:
@PeterKietzmann, @haukepetersen, how do we handle this in general? Didn't we agree on a failsafe counter for these kind of loops to avoid infinitely trying?